certbot/tests/compat/filesystem_test.py

```diff
@@ -6,41 +6,162 @@
 from certbot.tests.util import TempDirTestCase
 
 
-class FilesystemTest(TempDirTestCase):
-    """Unit tests for filesystem module."""
-    @unittest.skipIf(os.name != 'nt', reason='Test specific to Windows security')
+@unittest.skipIf(os.name != 'nt', reason='Test specific to Windows security')
+class WindowsChmodTests(TempDirTestCase):
+    """Unit tests for Windows chmod function in filesystem module"""
+    def test_symlink_resolution(self):
+        probe_path = _create_probe(self.tempdir)  # This is 0o744 by default
+
+        link_path = os.path.join(self.tempdir, 'link')
+        os.symlink(probe_path, link_path)
+
+        ref_dacl_probe = _get_security_dacl(probe_path).GetSecurityDescriptorDacl()
+        ref_dacl_link = _get_security_dacl(link_path).GetSecurityDescriptorDacl()
+
+        # Removing the rights for `all`, we have at least one ACL less than in the case of 0o744.
+        filesystem.chmod(link_path, 0o700)
+
+        # Assert the real file is impacted, not the link.
```

Comment:
Related to this point, do you have any good docs about how permissions on symlinks work on Windows?

Reply:
Official doc about DACL edition: https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/icacls, that explain the `/l` flag, which makes the operations on DACL be done for the symbolic link  instead of the target file.